src/kirby/components/icon/custom-icon-name.directive.tns.ts

```diff
@@ -0,0 +1,65 @@
+import { LayoutBase } from 'tns-core-modules/ui/layouts/layout-base';
+import { Label } from 'tns-core-modules/ui/label/label';
+import {
+  Directive,
+  ElementRef,
+  Input,
+  Inject,
+  AfterViewInit,
+  OnChanges,
+  SimpleChanges,
+} from '@angular/core';
+
+import { CUSTOM_FONT_SETTINGS, CustomIconSettings } from './custom-icon-settings';
+
+@Directive({
+  // tslint:disable-next-line:directive-selector
+  selector: '[customName]',
+})
+export class CustomIconNameDirective implements AfterViewInit, OnChanges {
+  // tslint:disable-next-line:no-input-rename
+  @Input('customName') customName: string;
+
+  constructor(
+    private element: ElementRef,
+    @Inject(CUSTOM_FONT_SETTINGS) private customIconSettings: CustomIconSettings
+  ) {}
+
+  ngAfterViewInit(): void {
+    this.handleCustomIcon(this.customName);
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.customName) {
+      this.handleCustomIcon(this.customName);
+    }
+  }
+
+  handleCustomIcon(name: string) {
+    if (name && this.customIconSettings[0].icons) {
+      const fontFamily = this.customIconSettings[0].fontfamily;
+      const icon = this.getCustomIcon(name);
+
+      if (fontFamily === undefined) {
+        console.warn('Missing font-family.');
+      } else if (icon === undefined) {
+        console.warn('Icon with name', name, 'was not found in custom font.');
+      } else {
+        this.setCustomIcon(icon.unicode, fontFamily);
+      }
+    }
+  }
+
+  getCustomIcon(name: string): { name: string; svg: string; unicode: string } {
```

Comment:
I find it nice to mark functions as private, just so the reader of the code clearly can see that they are not called anywhere beside the containing class. Not a major issue, but one you should consider.